wip-trampoline.md

```diff
@@ -25,6 +25,15 @@ but that theoretically allows another level of nesting (mind blown).
 However not really usable given the space constraints...
 
 TODO: maybe this should be version 1 of the overall onion version namespace?
+It can then also be used as a top-level onion (smaller than the v0 onion).
+It's like the v0 onion but lets each hop choose how they route to the next hop.
+If they have a direct channel they directly send the unwrapped v1 onion to the
+next hop.
+If they don't they wrap the v1 onion inside a v0 to get to the next hop via
+intermediate hops.
+Even though the onion packet is smaller, you don't know if previous hops used
+the direct route or went through v0 wrappings, so you have even greater doubts
+about your position in the route than with only v0.
```

Comment:
I disagree with the fact that v1 should be a trampoline onion. The onion construction we have is versatile enough to be reused as is for the trampoline onion.

The chances of that happening is virtually 0 and we shouldn't be optimizing for this case, and you are basically giving away to an observer that this was a trampoline onion and not a routing onion, by just observing a smaller packet go over the encrypted connection.

Reply:
This is an important point, this is a fundamental design choice. Let's discuss it in the global conversation for visibility (your comment was made on an old commit).